test/e2e/suite/command/inspect.go

```diff
@@ -0,0 +1,63 @@
+package command
+
+import (
+	. "github.com/notaryproject/notation/test/e2e/internal/notation"
+	"github.com/notaryproject/notation/test/e2e/internal/utils"
+	. "github.com/notaryproject/notation/test/e2e/suite/common"
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var inspectSuccessfully = []string{
+	"└── application/vnd.cncf.notary.signature",
+	"└── sha256:",
+	"├── media type:",
+	"├── signature algorithm:",
+	"├── signed attributes",
+	"signingTime:",
+	"signingScheme:",
+	"├── user defined attributes",
+	"│   └── (empty)",
+	"├── unsigned attributes",
+	"│   └── signingAgent: Notation/",
+	"├── certificates",
+	"│   └── SHA256 fingerprint:",
+	"issued to:",
+	"issued by:",
+	"expiry:",
+	"└── signed artifact",
+	"media type:",
+	"digest:",
+	"size:",
+}
+
+var _ = Describe("notation inspect", func() {
+	It("all signatures of an image", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			notation.Exec("inspect", artifact.ReferenceWithDigest()).
+				MatchKeyWords(inspectSuccessfully...)
+		})
+	})
+
+	It("all signatures of an image with TLS", func() {
+		HostInGithubAction(BaseOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.DomainReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			notation.Exec("inspect", artifact.DomainReferenceWithDigest()).
+				MatchKeyWords(inspectSuccessfully...)
+		})
+	})
+
+	It("all signatures of an image with --insecure-registry flag", func() {
+		HostInGithubAction(BaseOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.DomainReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			notation.Exec("inspect", "--insecure-registry", artifact.DomainReferenceWithDigest()).
+				MatchKeyWords(inspectSuccessfully...)
```

Comment:
How are we verifying communication indeed happened over https  and http in case of --insecure-registry usecase.

Reply:
You can read the testing debug log from  `Run e2e tests` step of `build` Github Actions

Reply:
For example: https://github.com/notaryproject/notation/actions/runs/5063834914/jobs/9090861035?pr=679

Reply:
yes but how are we validating this in test? For e.g. in future if someone introduce a bug where `--insecure-registry` is getting ignored and notation is using https, how will this test catch the issue?

Reply:
Got it. Updated to include the assertion for HTTP and HTTPS request.